include/csoundCore.h

```diff
@@ -1098,74 +1341,52 @@ typedef struct _message_queue_t_ {
     int (*GetMessageLevel)(CSOUND *);
     void (*SetMessageLevel)(CSOUND *, int messageLevel);
     void (*SetMessageCallback)(CSOUND *,
-                void (*csoundMessageCallback)(CSOUND *,
-                                              int attr, const char *format,
-                                              va_list valist));
-    /**@}*/
-    /** @name Event and MIDI functionality for opcodes */
-    /**@{ */
-    int (*SetReleaseLength)(void *p, int n);
-    MYFLT (*SetReleaseLengthSeconds)(void *p, MYFLT n);
-    int (*GetMidiChannelNumber)(void *p);
-    MCHNBLK *(*GetMidiChannel)(void *p);
-    int (*GetMidiNoteNumber)(void *p);
-    int (*GetMidiVelocity)(void *p);
-    int (*GetReleaseFlag)(void *p);
-    double (*GetOffTime)(void *p);
-    MYFLT *(*GetPFields)(void *p);
-    int (*GetInstrumentNumber)(void *p);
-    int (*GetZakBounds)(CSOUND *, MYFLT **);
-    int (*GetTieFlag)(CSOUND *);
-    int (*GetReinitFlag)(CSOUND *);
-    /** Current maximum number of strings, accessible through the strset
-        and strget opcodes */
-    int (*GetStrsmax)(CSOUND *);
-    char *(*GetStrsets)(CSOUND *, long);
-    /* Fast power of two function from a precomputed table */
-    MYFLT (*Pow2)(CSOUND *, MYFLT a);
-    /* Fast power function for positive integers */
-    MYFLT (*intpow)(MYFLT, int32);
-    /* Returns a string name for the file type */
-    char *(*type2string)(int type);
+                               void (*csoundMessageCallback)
+                               (CSOUND *,int attr, const char *format,
+                                va_list valist));
     /**@}*/
-    /** @name Arguments to opcodes */
+    
+    /** @name Arguments and Types */
     /**@{ */
-    CS_TYPE *(*GetTypeForArg)(void *p);
-    int (*GetInputArgCnt)(void *p);
-    char *(*GetInputArgName)(void *p, int n);
-    int (*GetOutputArgCnt)(void *p);
-    char *(*GetOutputArgName)(void *p, int n);
     char *(*GetString)(CSOUND *, MYFLT);
-    int32 (*strarg2insno)(CSOUND *, void *p, int is_string);
-    char *(*strarg2name)(CSOUND *, char *, void *, const char *, int);
-
+    int32 (*StringArg2Insno)(CSOUND *, void *p, int is_string);
+    char *(*StringArg2Name)(CSOUND *, char *, void *, const char *, int);
+    const CS_TYPE *(*StringType)(CSOUND *csound);
```

Comment:
Not for this PR, but I think if we're adding these here we could create and add these to a "StandardTypes" struct. Also in general, as CS7 progresses, we might benefit to partition CSOUND into more sub-types, brainstorming things like:

* csound.mem (memory)
* csound.sndfile (sound file I/O)
* csound.ftables
* etc.